### PR TITLE
Ignore modules, variables, functions when creating type reference

### LINKED ISSUE #362: Type reference resolves incorrectly when namespace and class have same name
```
declare namespace lunr {
    namespace Builder {
        type Plugin = (this: Builder, ...args: any[]) => void;
    }
    class Builder {
        
    }
}
```
Reference to `Builder` in `Plugin` will be resolved to namespace, not class, which will result in undefined FQN in this type.

### PR DESCRIPTION
**Pull request by trilis**
fixes #362

### CODE REVIEW

File: typescript/ts-converter/src/AstConverter.ts [hunk 1]
```diff
@@ -746,7 +742,10 @@ export class AstConverter {
     }
 
     if (Array.isArray(symbol.declarations)) {
-      return symbol.declarations[0];
+      return symbol.declarations.filter(decl => !ts.isModuleDeclaration(decl) &&
```
Comment (Schahen) [inline typescript/ts-converter/src/AstConverter.ts:745]: let's try to just find the first one that matches our criteria - the order is guaranteed
 by TypeScript (and if it won't, we'll catch it in tests)

File: compiler/test/data/typescript/mergeDeclarations/namespaceWith/class.d.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import org.w3c.notifications.*
+import org.w3c.performance.*
+import org.w3c.workers.*
+import org.w3c.xhr.*
+
+typealias Plugin = (self: Builder, args: Any) -> Unit
+
+// ------------------------------------------------------------------------------------------
+// [test] class.lunr.kt
+@file:JsQualifier("lunr")
```
Comment (Schahen) [inline compiler/test/data/typescript/mergeDeclarations/namespaceWith/class.d.kt:24]: if it is a name of actually existing package I'd rather rename

**Review by Schahen — COMMENTED** [review]
Apart from this two comments I don't have any other

File: compiler/test/data/typescript/mergeDeclarations/namespaceWith/class.d.kt (hunk 2 above)
Comment (trilis) [reply compiler/test/data/typescript/mergeDeclarations/namespaceWith/class.d.kt:24]: Sure

**Pull request merged**